errors.go

```diff
@@ -36,3 +27,33 @@ var (
 	// due to floating-point arithmetic.
 	ErrNoProgress = errors.New("linesearch: no change in location after Linesearcher step")
 )
+
+// ErrFuncInit is returned when an initial function value is invalid. The error
+// state may be either +Inf or NaN. ErrFuncInit satisfies the error interface.
+type ErrFuncInit float64
```

Comment:
Which is better, `ErrFuncInit` or `ErrInitFunc`? The latter sounds better to me.